index.js

```diff
@@ -1,14 +1,29 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
+module.exports = function (target, readonly) {
```

Comment:
> but like this is better, on a side node, i called the method decorateWith(arg), seemed more semantic

That doesn't make much sense when you think about how it's going to be used:

```js
@known.decorateWith
class Foobar {
    baz = 1;
    someMethod() {}
}
```

This looks better:

```js
@known.decorator
class Foobar {
    baz = 1;
    someMethod() {}
}
```